drivers/gpio/gpio_shell.c

```diff
@@ -454,21 +454,22 @@ static void print_gpio_ctrl_info(const struct shell *sh, const struct gpio_ctrl
 {
 	gpio_pin_t pin;
 	bool reserved;
+	const char *line_name;
 
 	shell_print(sh, " ngpios: %u", ctrl->ngpios);
 	shell_print(sh, " Reserved pin mask: 0x%08X", ctrl->reserved_mask);
 
 	shell_print(sh, "");
 
 	shell_print(sh, " Reserved  Pin  Line Name");
-	for (pin = 0; pin < GPIO_MAX_PINS_PER_PORT; pin++) {
-		if ((pin >= ctrl->ngpios) && (pin >= ctrl->line_names_len)) {
```

Comment:
It looks like it should have been an `||` instead of a `&&` here. The fix also improves the output when there are fewer names than pin.